beacon_node/network/src/router/processor.rs

```diff
@@ -88,8 +88,14 @@ impl<T: BeaconChainTypes> Processor<T> {
 
     /// An error occurred during an RPC request. The state is maintained by the sync manager, so
     /// this function notifies the sync manager of the error.
-    pub fn on_rpc_error(&mut self, peer_id: PeerId, request_id: RequestId) {
-        self.send_to_sync(SyncMessage::RPCError(peer_id, request_id));
+    pub fn on_rpc_error(&mut self, peer_id: PeerId, request_id: Option<RequestId>) {
+        match request_id {
+            Some(request_id) => self.send_to_sync(SyncMessage::RPCError(peer_id, request_id)),
+            None => crit!(
```

Comment:
Yeah, these are the result of us not creating 4 types for each case. I can live with this if you can

Reply:
I think it's fine tbh

Reply:
Let me track this down and I'll get back to you.

Reply:
So, this means that `RPCCodedResponse`s are always sent from our side?
If that's the case, I do prefer to have the different types because this is just noise

Reply:
I'm not sure about this, but if this noise can be avoided I would go with that, so I just marked as work in progress again to go decide something better

Reply:
I went on and added separate types for all four of them. and this stayed. Errors are only produced on outbound requests, so they are assigned to the original request id, which is an option

Reply:
Yeah. I like the extra types. They make things much easier to understand. 
We could still fix this which wouldn't be too hard, now that you've done the extra types. The Requests that don't have an ID are PING and MetaData and they all are handled within the behaviour and are never exposed outside. 
So what we should do is change `BehaviourEvent::RPC(_ ,Received)` to a new type that doesn't have an option, as these are the only ones that will be emitted from the behaviour. 
That should clean up these cases ensuring we never reach these code paths.

Reply:
> We could still fix this which wouldn't be too hard

it was harder than expected,

but solved